eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -48,6 +48,9 @@ case class WaitingForRevocation(nextRemoteCommit: RemoteCommit, sent: CommitSig,
 // @formatter:on
 
 trait AbstractCommitments {
+  /** Specifically return incoming HTLCs which were cross-signed AND relayed into another local channel. */
+  def getIncomingCrossSignedAndRelayedHtlcs: Set[UpdateAddHtlc]
```

Comment:
I think it's confusing that it's named that way.
I think it would be better to expose in this trait:

```
/** Returns all HTLCs in the current remote commitment. */
def htlcsRemoteCommit: Set[DirectedHtlc]
/** Returns all HTLCs in the current local commitment. */
def htlcsLocalCommit: Set[DirectedHtlc]
```

And then you compose the results with the usual `outgoing` / `incoming` partial functions.